### PR TITLE
Fix FileUploadCleanupFilterAttribute

### PR DESCRIPTION
**Pull request by ronaldbarendse**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/6827.

### Description
The `FileUploadCleanupFilterAttribute` only did null-checks on `actionExecutedContext.Request.Content`, but then used `actionExecutedContext.Response.Content`.

This fixes all the checks and:
- Removes redundant logging: if there's nothing we can do, just skip cleaning up the uploaded files (the log messages do not provide real help anyway)
- Check all action arguments for `IHaveUploadedFiles` models (not only the first)
- Remove unused folder tracking

We could use the `actionExecutedContext.Response.TryGetContentValue<T>()` extension method instead of getting the `ObjectContent` and then the `Value`, but the current logic will probably be exactly the same...

When testing, the following should still work:
- Upload of media, media on content, local package install, document type import, user avatar (current or other), member import and blueprint/content template import
- Temporary files should be cleaned up (`App_Data\Temp\FileUploads`)

### CODE REVIEW

File: src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs [hunk 1]
```diff
@@ -1,149 +1,85 @@
...
+                    // Cleanup any files associated
+                    foreach (var uploadedFile in haveUploadedFiles.UploadedFiles)
+                    {
+                        try
+                        {
+                            File.Delete(uploadedFile.TempFilePath);
+                        }
+                        catch (Exception ex)
+                        {
+                            Current.Logger.Error<FileUploadCleanupFilterAttribute>(ex, "Could not delete temp file {FileName}", uploadedFile.TempFilePath);
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs]: Should this be logged as an error? Maybe a warning in more appropriate?

File: src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs [hunk 2]
```diff
@@ -1,149 +1,85 @@
...
-                        }
+                    try
+                    {
+                        File.Delete(uploadedFile.TempFilePath);
                     }
-                    else
+                    catch (Exception ex)
                     {
-                        Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The actionExecutedContext.Request.Content.Value is not IHaveUploadedFiles, it is {ObjectType}", objectContent.Value.GetType());
+                        Current.Logger.Error<FileUploadCleanupFilterAttribute>(ex, "Could not delete temp file {FileName}", uploadedFile.TempFilePath);
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs]: Same here: probably better to log this as a warning...

File: src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs [hunk 3]
```diff
@@ -1,149 +1,85 @@
...
                     }
-                }
-                else
-                {
-                    Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The actionExecutedContext.Request.Content is not ObjectContent, it is {RequestObjectType}", actionExecutedContext.Request.Content.GetType());
+
+                    // Clear out the temp path, so it's not returned in the response
+                    uploadedFile.TempFilePath = null;
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs]: Clearing out should set this to the initial value of `null` (same as when initiating a new `ContentPropertyFile`).

**Comment by emmaburstow** [comment]
Morning @ronaldbarendse,

Thanks for the PR. We'll take a look at your questions and get an answer to you soon. 

Emma

**Pull request closed without merging**